flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetTumbleTimeWindowAggReduceCombineFunction.scala

```diff
@@ -38,19 +40,35 @@ import org.apache.flink.types.Row
   * @param keysAndAggregatesArity The total arity of keys and aggregates
   */
 class DataSetTumbleTimeWindowAggReduceCombineFunction(
-    genAggregations: GeneratedAggregationsFunction,
+    genPreAggregations: GeneratedAggregationsFunction,
+    genFinalAggregations: GeneratedAggregationsFunction,
     windowSize: Long,
     windowStartPos: Option[Int],
     windowEndPos: Option[Int],
     keysAndAggregatesArity: Int)
   extends DataSetTumbleTimeWindowAggReduceGroupFunction(
-    genAggregations,
+    genFinalAggregations,
     windowSize,
     windowStartPos,
     windowEndPos,
     keysAndAggregatesArity)
     with CombineFunction[Row, Row] {
 
+  protected var preAggfunction: GeneratedAggregations = _
```

Comment:
We need two code-gen'd functions here as well. The only difference is in the `setAggregationResults()` method. The partial function returns the accumulators, the final one the result of `AggregateFunction.getValue()`.

Alternatively, we could add a method `setAccumulators()` method to `GeneratedAggregations` which sets the accumulators.

What do you think?